File: capstone/src/main/java/impact/capstone/user/controller/UserController.java
package impact.capstone.user.controller;

import impact.capstone.user.model.dto.UserDTO;
import impact.capstone.user.service.UserService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/users")
public class UserController {
    UserService userService;

    @Autowired
    public UserController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping("/signup")
    @Operation(summary = "회원가입", description = "유저 정보를 입력받아 회원가입 하는 기능")
    @ApiResponses(value = {
            @ApiResponse(responseCode = "200", description = "회원가입 성공"),
            @ApiResponse(responseCode = "400", description = "ID가 중복입니다.")
    })
    public ResponseEntity<UserDTO> signup(@RequestBody UserDTO userDTO) {
        return userService.signUp(userDTO);
    }

    @PostMapping("/login")
    @Operation(summary = "로그인", description = "유저 정보를 입력받아 로그인하는 기능")
    @ResponseBody
    public ResponseEntity<String> signIn(@RequestBody UserDTO userDTO) {
        return userService.signIn(userDTO.getId(), userDTO.getPassword());
    }

    @PostMapping("/updateuserinfo")
    @Operation(summary = "회원 정보 수정", description = "회원 정보를 수정한다")
    public ResponseEntity<UserDTO> updateAccount(
            @Parameter(name = "JWT", description = "jwt")
            @RequestHeader(value = "jwt") String token,
            @RequestBody UserDTO userDTO) {
        return userService.updateAccount(userDTO, token);
    }

    @PostMapping("/deleteuser")
    @Operation(summary = "회원 탈퇴", description = "회원 정보를 삭제한다")
    public ResponseEntity<String> deleteAccount(
            @Parameter(name = "JWT", description = "유저 토큰")
            @RequestHeader(value = "jwt") String token) {
        return userService.deleteAccount(token);
    }
}

File: server/src/main/kotlin/deu/capstone/impact/server/controller/UserController.kt
package deu.capstone.impact.server.controller

import deu.capstone.impact.server.dto.user.LoginRequestDto
import deu.capstone.impact.server.dto.user.LoginResponseDto
import deu.capstone.impact.server.service.JwtService
import deu.capstone.impact.server.service.UserService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/user")
class UserController(
        private val userService: UserService,
        private val jwtService: JwtService,
) {
    /**
     * 로그인을 위한 api
     * @param 로그인을 시도한 유저 정보
     * @return jwt / 실패 시 에러 메시지
     */
    @PostMapping("/login")
    fun login(@RequestBody request: LoginRequestDto): LoginResponseDto {
        val user = userService.login(request)

        val token = jwtService.generateToken(user)

        return LoginResponseDto(
                token = token
        )
    }
}